backend/src/users/auth.service.ts

```diff
@@ -0,0 +1,176 @@
+
+    const hashPassword = await bcryptjs.hash(password, bcryptOptions.salt);
+
+    const newUser = await this.usersService.create(email, hashPassword, organizationId);
+
+    const { access_token, refresh_token } = await this.getTokens(newUser.id, newUser.email);
+
+    await this.updateRtHash(newUser.id, refresh_token);
+
+    await this.sendActivationEmail(newUser.id, newUser.email);
+
+    return { access_token, refresh_token, organizationId };
+  }
+
+  async login(email: string, password: string, res) {
+    const [user] = await this.usersService.find(email);
+
+    if (!user) {
+      throw new NotFoundException('user not found');
+    }
+
+    const isCorrectPass = await bcryptjs.compare(password, user.password);
+    if (!isCorrectPass) throw new ForbiddenException('Access Denied');
+
+    const { access_token, refresh_token } = await this.getTokens(user.id, user.email);
+
+    const { id: organizationId } = await this.organizationsService.findOne(user.organizationId);
+
+    if (!organizationId) {
+      throw new NotFoundException('organization not found');
+    }
+
+    res.cookie('refreshToken', refresh_token, {});
+    res.cookie('accessToken', access_token, {});
+
+    await this.updateRtHash(user.id, refresh_token);
+
+    return res.send({ access_token, refresh_token, organizationId });
+  }
+
+  async refreshTokens(userId: number, rt: string): Promise<Tokens> {
+    const user = await this.usersService.findOne(userId);
+
+    if (!user || !user.hashedRt) {
+      throw new ForbiddenException('Access Denied at user');
+    }
+
+    const isCorrectRT = await bcryptjs.compare(rt, user.hashedRt);
+
+    if (!isCorrectRT) {
+      throw new ForbiddenException('Access Denied for you at matcher');
+    }
+
+    const tokens = await this.getTokens(user.id, user.email);
+    await this.updateRtHash(user.id, tokens.refresh_token);
+
+    return tokens;
+  }
+
+  async getTokens(userId: number, email: string) {
+    const jwtOptions = await this.config.get('jwt');
+    const [at, rt] = await Promise.all([
+      this.jwtService.signAsync(
+        {
+          sub: userId,
+          email,
+        },
+        {
+          secret: 'Secret',
+          expiresIn: '30m',
+        }
+      ),
+      this.jwtService.signAsync(
+        {
+          sub: userId,
+          email,
+        },
+        {
+          secret: 'rt-secret',
+          expiresIn: '1d',
+        }
+      ),
+    ]);
+
+    return {
+      access_token: at,
+      refresh_token: rt,
+    }
+  }
+
+  async updateRtHash(userId: number, rt: string) {
+    const bcryptOptions = await this.config.get('bcryptConf');
+    const hash = await bcryptjs.hash(rt, bcryptOptions.salt);
+    await this.usersService.update(userId, {
+      hashedRt: hash,
+    })
+  }
+
+  async decodeToken(token: string): Promise<number> {
+    const decoded = await this.jwtService.verifyAsync(token, { secret: 'Secret' });
+    const { sub } = decoded;
+
+    return sub;
```

Comment:
A redundant destructurization. 
```suggestion
    return decoded.sub;
```